weave/integrations/langchain_nvidia_ai_endpoints/langchain_nv_ai_endpoints.py

```diff
@@ -0,0 +1,221 @@
+from __future__ import annotations
+
+import importlib
+import time
+from typing import Any, Callable
+
+import_failed = False
+
+try:
+    from langchain_core.messages import AIMessageChunk, convert_to_openai_messages
+    from langchain_core.outputs import ChatGenerationChunk, ChatResult
+except ImportError:
+    import_failed = True
+
+import weave
+from weave.trace.autopatch import IntegrationSettings, OpSettings
+from weave.trace.op import Op, ProcessedInputs
+from weave.trace.op_extensions.accumulator import add_accumulator
+from weave.trace.patcher import MultiPatcher, NoOpPatcher, SymbolPatcher
+
+_lc_nvidia_patcher: MultiPatcher | None = None
+
+
+# NVIDIA-specific accumulator for parsing the objects of streaming interactions
+def nvidia_accumulator(acc: Any | None, value: Any) -> Any:
+    if acc is None:
+        acc = ChatGenerationChunk(message=AIMessageChunk(content=""))
+    acc = acc + value
+
+    ## Need to do this since the __add__ impl for the streaming response is wrong
+    ## We will get the actual usage in the final chunk so this will be eventually consistent
+    acc.message.usage_metadata = value.message.usage_metadata
+
+    return acc
+
+
+# Post processor to transform output into OpenAI's ChatCompletion format -- need to handle stream and non-stream outputs
+def postprocess_output_to_openai_format(output: Any) -> dict:
+    """
+    Need to post process the output reported to weave to send it on openai format so that Weave front end renders
+    chat view. This only affects what is sent to weave.
+    """
+    from openai.types.chat import ChatCompletion
+
+    if isinstance(output, ChatResult):  ## its ChatResult
```

Comment:
nit: single hash for comments (throughout this file)

Reply:
addressed